cmd/oras/repository/list.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repository
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"oras.land/oras/cmd/oras/internal/option"
+)
+
+type repositoryOptions struct {
+	option.Remote
+	option.Common
+	hostname string
+}
+
+func listCmd() *cobra.Command {
+	var opts repositoryOptions
+	cmd := &cobra.Command{
+		Use:   "list REGISTRY [flags]",
+		Short: "[Preview] List the repositories under the registry",
+		Long: `[Preview] List the repositories under the registry
+** This command is in preview and under development. **
+Example - Fetch raw manifest:
+  oras repository list localhost:5000
+`,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.hostname = args[0]
+			return listRepository(opts)
+		},
+	}
+
+	option.ApplyFlags(&opts, cmd.Flags())
+	return cmd
+}
+
+func listRepository(opts repositoryOptions) error {
+	ctx, _ := opts.SetLoggerLevel()
```

Comment:
This might be a personal preference so its non-blocking. I think most of the lines are probably self-explanatory so maybe you can remove the comments in the function. Just makes code easier to read :)

Reply:
I will improve the piece of code and remove the comments 😊